frontend/src/api/prometheus/__tests__/usePrometheusQueryRange.spec.ts

```diff
@@ -0,0 +1,121 @@
+import axios from 'axios';
+import { act } from '@testing-library/react';
+import { testHook } from '~/__tests__/unit/testUtils/hooks';
+import { mockPrometheusServing } from '~/__mocks__/mockPrometheusServing';
+import usePrometheusQueryRange from '~/api/prometheus/usePrometheusQueryRange';
+
+jest.mock('axios', () => ({
+  post: jest.fn(),
+}));
+
+const mockAxios = jest.mocked(axios.post);
+
+describe('usePrometheusQueryRange', () => {
+  const responsePredicate = jest.fn();
```

Comment:
Need a test for `responsePredicate` being called and asserting the expected values are being passed into the function and that the output of the function influences the output of the hook.